### ISSUE TITLE
NetworkAccessManager outdated

### ISSUE THREAD

**Issue by molakos** [rduivenvoorde/pdokservicesplugin#48]
QGIS version: 3.22

Hello there!

The official documentation suggested to use `NetworkAccessManager` to wrap custom http requests which requires custom authorization (in my case is oauth2).

While I was trying to make my script work I started to face an error message:

```
/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\pdokservicesplugin\networkaccessmanager.py", line 190, in request
QgsAuthManager.instance().updateNetworkRequest(req, self.authid)
AttributeError: type object 'QgsAuthManager' has no attribute 'instance'
```

I did some digging and found this [issue](https://issues.qgis.org/issues/17384) on QGIS's issue tracker. It looks like that the core code was updated and they have removed the `instance` method from `QgsAuthManager` with the following comment: 

`instance() was removed in favor of QgsApplication::authManager() that returns an instance of the Auth Manager`

**Comment by rduivenvoorde** [comment]
@molakos the networkservicesaccessmanager.py is actually an old helper script (not specific of this plugin) to be removed in a next version in favour of normal QgsNetworkManager calls.
The plugin itself does not have any authenticated layers (is it?), but feel free to create a Pull Request if you think it would be better to have this fixed.
Thanks for the heads up.

**Comment by molakos** [comment]
@arbakker thank you for your input! 

In some ways our company does use PDOK services, but the code above which I highlighted was mentioned in the original documentation of pyQGIS and this is how I found your code snippet.

Since we are kind of new in the QGIS plugin development, I was simply looking around for authentication solutions, more precisely on how to actually make an authorized GET request to our own custom API (which uses Auth0 tokens). 

In the meantime I was able to tackle the problem. I believe we can close this issue :)

Thank you for helping me out and also for your nice services! It helped me a lot.
